### DISCUSSION TITLE
setting reaction requiresObservable: false could turn off warnings

### DISCUSSION THREAD

**Discussion by KevinBrogan** [mobxjs/mobx#3313] [Ideas]
I can setup a reaction and pass it the options `{requiresObservable:true}` to turn on the config warning for just that one reaction, instead of turning them all on with `configure({reactionRequiresObservable: true})`, but I can't do the opposite.

I would like to be able turn off the global config on a case by case basis.

My use case is dealing with user generated code, and the user may decide to not output an observable property value. I can't leave warnings on for my code and off for user code but that would be very useful.

**Comment by urugator** [comment]
Similar to #3214. I will take a look at it once I have a sec.

**Comment by KevinBrogan** [comment]
Thanks! Apparently I don't know how to use github to search. I did a search
for requiresReaction_ and nothing came up.

I took a look at https://github.com/mobxjs/mobx/pull/3214/files and it's
basically the ts script version of the local changes that I made myself.

He's chosen to use a conditional check on the type of
this.requiresReaction_.
Personally, I prefer a nullish coalescing operator, as in the second screen
shot, although I didn't edit the original typeScript. I edited the compiled
javascript for my local copy.

I've not had a reason to look at overriding any of the other configuration
options, but if overrides exist elsewhere, it would be good to make all
options consistently overridable.

Cheers,
Kevin

.
[image: image.png]
[image: image.png]

**Reply by urugator** [reply]
> He's chosen to use a conditional check on the type

Because if you would pass (intentionally or not) anything non-nullish, eg `{ requiresReaction: object }`, it would have the same effect as `true` and I want to avoid such ambiguity, because then it wouldn't be clear whether eg `null`/`undefined`/`0` should similary coerce to `false` or if it means unspecified.

>  second screen shot

images are not visible

**Comment by urugator** [comment]
Fixed in 6.4.2
